services/fileops/fileops.go

```diff
@@ -0,0 +1,206 @@
+	var jobs [][]string
+	if err != nil {
+		fmt.Println("error Openning the file concurrently" + err.Error())
+	}
+
+	csvReaderCon := csv.NewReader(csvFile)
+	for {
+		rStr, err := csvReaderCon.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			fmt.Println("ERROR: ", err.Error())
+			break
+		}
+		jobs = append(jobs, rStr)
+	}
+	return jobs
+}
+
+//UpdateFile the CSV from the target API
+func (s *FileOpsService) UpdateFile(jsonData []model.Joke) {
+	pwd, err := os.Getwd()
+	csvFile, err := os.OpenFile(pwd+"/data/data.csv", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		fmt.Println(err)
+	}
+	defer csvFile.Close()
+	records, chkRecords := readCsvFile(pwd + filePath)
+	if len(chkRecords) != 0 {
+		fmt.Println("Error Updating file ")
+	}
+
+	counter := len(records) + 1
+	writer := csv.NewWriter(csvFile)
+
+	for i, usance := range jsonData {
+		var row []string
+		row = append(row, strconv.Itoa(counter+i))
+		row = append(row, strings.Replace(usance.Setup, "\n", "", -1))
+		row = append(row, usance.Punchline)
+		writer.Write(row)
+	}
+	writer.Flush()
+	s.LoadData()
+}
+
+//LoadData this will load the data from the csv
+func (s *FileOpsService) LoadData() string {
+	var message string
+	allJokes = append(allJokes[:0])
+	pwd, _ := os.Getwd()
+	csvFile, err := os.Open(pwd + filePath)
+	if err != nil {
+		message = "Error openning file: " + err.Error()
+	}
+	defer csvFile.Close()
+	r := csv.NewReader(csvFile)
+	records, err := r.ReadAll()
+	if err != nil {
+		fmt.Println(err)
+		message = "Error reading file: " + err.Error()
+		return message
+	}
+	if len(records) == 0 {
+		message = "Currently there are not Jokes Stored... this is not funny!"
+		return message
+	}
+	for _, rec := range records {
+		var insert model.Joke
+		insert.ID, err = strconv.Atoi(rec[0])
+		insert.Setup = rec[1]
+		insert.Punchline = rec[2]
+		allJokes = append(allJokes, insert)
+	}
+	log.Println("[Info] Successfully loaded jokes from file")
+	return message
+}
+
+//ReadData will return the data from the file
+func (s *FileOpsService) ReadData() []model.Joke {
+	log.Println("[Info] Returned Jokes, a total of ", len(allJokes))
+	return allJokes
+}
+
+func (s *FileOpsService) ReadConcurrently(typeOfInt string, items int, itemsPerWorker int) []*model.Joke {
+
+	conJoke := make(chan *model.Joke)        // Chanel to return the joke from the worker
+	numWorkerPools := items / itemsPerWorker //Number of Worker pools, result will go to floor
+	log.Println("Workers for the requests: " + strconv.Itoa(numWorkerPools))
+	killSignal := make(chan struct{})               // Kill signal
+	var finalRetJokes []*model.Joke                 // final return of values
+	dataFromFile := readCsvFileConcurrent(filePath) // Load the data from the file
+	log.Println(numWorkerPools)
+	var wg sync.WaitGroup // Control for the pool
+	// Operation to perform with the worker
+	worker := func(results chan<- *model.Joke) {
+		for {
+
+			n := rand.Intn(len(dataFromFile))
```

Comment:
Could it be possible that we get duplicate data by doing this?

Reply:
Yes, it is possible, good catch! Added validation to keep them unique